src/core/xml_parser.c

```diff
@@ -261,6 +262,21 @@ int xml_parser_get_attribute_enum(const char *key,
     return start_offset - 1;
 }
 
+resource_type xml_parser_get_resource(const char *key)
+{
+    const char *value = get_attribute_value(key);
+    if (!value) {
+        return RESOURCE_NONE;
+    }
+    for (resource_type i = RESOURCE_MIN; i < RESOURCE_MAX; i++) {
+        const char *resource_name = resource_get_data(i)->xml_attr_name;
+        if (resource_name != NULL && compare_multiple(resource_name, value)) {
```

Comment:
I don't think there is any case where the resource xml attr isn't set. Also, use `0`, not `NULL` please.